server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -1111,17 +1111,30 @@ public void registerDynamicRoute(RestHandler.Route route, RestSendToExtensionAct
             requireNonNull(route, "route is required");
             requireNonNull(action, "action is required");
             Optional<String> routeName = Optional.empty();
+            Set<String> actionNames = null;
```

Comment:
Initialize this to `new Hashset<String>()` then you can remove all the null checks for actionNames

Reply:
On 1121, replace this with `addAll`

Reply:
`.addAll(null)` will throw an NPE when `nr.actionNames()` is null. So I think its best to reassign on 1121 instead of addAll

Reply:
Why not fix `nr.actionNames()` so it is always empty list, much easier to handle when you are sure it isn't null

Reply:
I know this is a little pedantic, but these action names determine if requests are authorized or not - huge stakes if there is mistakes.  The fewer branching statements and the clearer the code the less chance of a CVE

Reply:
fair point.. updated the code